File: Tree/sum of left leaves/Left_sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) 
    {
        int val=0;
        if(root==NULL)
        {
            return 0;
        }
        if(root->left)
        {
            if(root->left->left==NULL and root->left->right ==NULL)
            {
                val = root->left->val;
            }
        }
        return val + sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right);
    }
};

File: Tree/sum of left leaves/Left_sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    sum =0
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def sumleft(root)->int:
            if root.left:
                if root.left.left is None and root.left.right is None:
                    return root.left.val
            return 0

        if root is None:
            return 0
        return sumleft(root)+self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right) 
        

       
            


        